### PR TITLE
fix(lightning): update transfer UI

### LINKED ISSUE #1288: [Bug]: In-flight HTLC
### Describe the bug

My channel got force closed while there was an in-flight HTLC. Though the closing transaction has many confirmations, the payment that was in-flight is still showing as pending under the spending balance, and the Lightning suggestion card is still showing "transferring".

IMG_2230
IMG_2231
IMG_2232
IMG_2233

### Screenshots / Recording

[bitkit_ldk_logs_1696301903578.zip](https://github.com/synonymdev/bitkit/files/12788045/bitkit_ldk_logs_1696301903578.zip)

### Operating system

iOS 17

### Bitkit version

72

### Log output

_No response_

### LINKED ISSUE #1335: [Bug]: Activity update
### Describe the bug

The chan closes don't show in activity

Screenshot 2023-10-17 at 11 02 18

https://synonymworkspace.slack.com/archives/C01H106JRQB/p1697533238125939?thread_ts=1697529943.255019&cid=C01H106JRQB

### Screenshots / Recording

_No response_

### Operating system

iOS

### Bitkit version

v82

### Log output

_No response_

### LINKED ISSUE #1484: [Bug]: Channel closing UX
I closed a channel locally, and my reserve was already sweeped to my on-chain wallet, but the sweep is still shown as pending. (ps: channel was opened via the Bitfinex and Blocktank integration).
image
image
image
image

I later opened and closed another channel, and it happened again (fortunately, the 4900 sats sweeped from the previous channel is no longer appearing as "pending"):
IMG_3C36AE238235-1

### PR DESCRIPTION
**Pull request by pwltr**
### Description

Updates the UI to better show transfers from and to spending, also contains most of the design updates from v46 related to transfers. I moved some state related to transfers to its own field in `wallet.transfers`, some of this is duplicated but this makes it more explicit and way easier to work with.

### Linked Issues/Tasks

Add any links to GitHub issues or Asana tasks that are relevant to this pull request.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [x] Unit test
- [ ] No test

### Screenshot / Video

channel open:

https://github.com/synonymdev/bitkit/assets/8538369/e798f87a-b2c1-45a0-b0e8-ea7b3e1cf07e

channel close:

https://github.com/synonymdev/bitkit/assets/8538369/b01c3e5a-8413-410c-ae4a-58147a5e029a

### CODE REVIEW

**Comment by limpbrains** [comment]
Not sure if this needs to be fixed in this ticket, but:
After I've initiated transfer to savings on Transfer screen slider is still shows that I have a channel

image

**Comment by limpbrains** [comment]
Also this error appeared: 
```
 ERROR  Error in setWalletData: [TypeError: Cannot read property 'height' of undefined]
 LOG  [TypeError: Cannot read property 'height' of undefined]
 ```

File: src/store/actions/wallet.ts [hunk 1]
```diff
@@ -296,6 +301,78 @@ export const injectFakeTransaction = ({
...
+	const transactions = currentWallet.transactions[selectedNetwork];
+	const transfers = currentWallet.transfers[selectedNetwork];
+	const pendingTransfers = transfers.filter((t) => {
+		return (
+			t.type !== ETransferType.open && t.status === ETransferStatus.pending
+		);
+	}) as TTransferToSavings[];
+
+	pendingTransfers.forEach((transfer) => {
+		const tx = transactions[transfer.txId];
```
Comment (limpbrains) [inline src/store/actions/wallet.ts:356]: Here, tx is undefined for me
Probably because I've initiated 2 transfers.
Sorry for spamming

**Comment by limpbrains** [comment]
> Not sure if this needs to be fixed in this ticket, but: After I've initiated transfer to savings on Transfer screen slider is still shows that I have a channel

Same happend when I've closed LN connection. After tx was confirmed and transfer was complete, I've opened Transfer flow and slider was not on the 0 position.

After app restart everything was correct

**Comment by pwltr** [comment]
> After I've initiated transfer to savings on Transfer screen slider is still shows that I have a channel

I think that needs to be addressed in RN-LDK, see https://github.com/synonymdev/bitkit/issues/1261

I'll try to fix the other findings

File: src/store/actions/wallet.ts (hunk 1 above)
Comment (pwltr) [reply src/store/actions/wallet.ts:356]: this should never be undefined so not sure what happened in your case but adding a check just in case.

**Comment by limpbrains** [comment]
Can we fix this one ot it also requiers 1261 ?

https://github.com/synonymdev/bitkit/assets/155891/6b049920-d53f-407a-ae7b-64ff7ec291d4

**Comment by pwltr** [comment]
> Can we fix this one ot it also requiers 1261 ?

It should open the initial LN flow now instead of the transfer flow.

**Review by limpbrains — APPROVED** [review]

**Comment by JeanlChristophe** [comment]
tested: OK

**Pull request merged**